pkg/simulator/objects/apply.go

```diff
@@ -0,0 +1,385 @@
+		} else {
+			dr = o.dc.Resource(restMapping.Resource)
+		}
+
+		resp, err = dr.Get(o.ctx, unstructuredObj.GetName(), metav1.GetOptions{})
+		var skipPatchStatus bool
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				resp, err = dr.Create(o.ctx, unstructuredObj, metav1.CreateOptions{})
+				if err != nil {
+					logrus.Errorf("error during creation of resource %s with gvr %s", unstructuredObj.GetName(), restMapping.Resource.String())
+					logrus.Error(unstructuredObj.Object)
+					o.addToFailedObjects(unstructuredObj, err)
+					// no need to patch status when object errors out
+					skipPatchStatus = true
+				}
+			} else {
+				return fmt.Errorf("error looking up object before creating the same %v", err)
+			}
+		}
+
+		if patchStatus && !skipPatchStatus {
+			// we will patch the status here later
+			status, ok, err := unstructured.NestedFieldCopy(unstructuredObj.Object, "status")
+			if err != nil {
+				return fmt.Errorf("error looking for status field: %v", err)
+			}
+			if ok {
+				/*processedObj, err := dr.Get(o.ctx, resp.GetName(), metav1.GetOptions{})
+				if err != nil && !apierrors.IsNotFound(err) {
+					return fmt.Errorf("error looking up resource %s with gvr %v with error %v", unstructuredObj.GetName(), unstructuredObj.GroupVersionKind(), err)
+				}*/
+				err = unstructured.SetNestedField(resp.Object, status, "status")
+				if err != nil {
+					return err
+				}
+				_, err = dr.UpdateStatus(o.ctx, resp, metav1.UpdateOptions{})
+				// update sometimes returns this object not found error
+				// the 404 lookup is to try and work around the same.
+				if err != nil && !apierrors.IsNotFound(err) {
+					o.addToFailedObjects(unstructuredObj, err)
+				}
+			}
+
+		}
+	}
+	return nil
+}
+
+// objectHousekeeping will perform some common housekeeping tasks based on GVR
+// needed to keep the apiserver happy since we are dealing with exported CRDs
+func objectHousekeeping(obj *unstructured.Unstructured) error {
+	// Common housekeeping performed on all objects
+	// need to clear resource version before apply.
+	// this will be added as an annotation
+	annotations, ok, err := unstructured.NestedStringMap(obj.Object, "metadata", "annotations")
+	if err != nil {
+		return err
+	}
+	if !ok {
+		annotations = make(map[string]string)
+	}
+
+	metadataMap, ok, err := unstructured.NestedMap(obj.Object, "metadata")
+	if err != nil {
+		return err
+	}
+
+	if ok {
+		orgCreationTimestamp, innerOK := metadataMap["creationTimestamp"]
+		if innerOK && orgCreationTimestamp != nil {
+			annotations[simCreationTimeStamp] = orgCreationTimestamp.(string)
+			unstructured.RemoveNestedField(obj.Object, "metadata", "resourceVersion")
+			err = unstructured.SetNestedStringMap(obj.Object, annotations, "metadata", "annotations")
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	switch obj.GetKind() {
+	//case "Ingress":
+	// Ingress specific housekeeping
+	//	err = ingressCleanup(obj)
```

Comment:
Why are these lines commented?

Reply:
Removed the lines. Left over from using a newer version of k8s in the initial version, which mean v1beta1 ingress objects needed a bit of housekeeping.